lib/AST/ASTContext.cpp

```diff
@@ -3594,7 +3594,6 @@ GenericFunctionType *GenericFunctionType::get(GenericSignature sig,
                                               Optional<ExtInfo> info) {
   assert(sig && "no generic signature for generic function type?!");
   assert(!result->hasTypeVariable());
-  assert(!result->hasPlaceholder());
```

Comment:
Could you add an assert that verifies that there are placeholder no placeholder that originated from a type variable directly contain one (e.g. dependent member types with type variable base)?